File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/MatchedAmountDiffFunction.java
package cz.fb.manaus.reactor.profit.progress.function;

import cz.fb.manaus.core.model.SettledBet;
import org.springframework.stereotype.Component;

import java.util.OptionalDouble;

@Component
public class MatchedAmountDiffFunction implements ProgressFunction {

    @Override
    public OptionalDouble function(SettledBet bet) {
        double actionAmount = bet.getBetAction().getPrice().getAmount();
        double betAmount = bet.getPrice().getAmount();
        return OptionalDouble.of(actionAmount - betAmount);
    }

}

File: mns-reactor/src/main/java/cz/fb/manaus/reactor/profit/progress/function/MatchedAmountDiffFunction.kt
package cz.fb.manaus.reactor.profit.progress.function

import cz.fb.manaus.core.model.RealizedBet
import org.springframework.stereotype.Component

@Component
object MatchedAmountDiffFunction : ProgressFunction {

    override fun invoke(bet: RealizedBet): Double {
        val actionAmount = bet.betAction.price.amount
        val betAmount = bet.settledBet.price.amount
        return actionAmount - betAmount
    }

}
